tests/nn/test_module.py

```diff
@@ -249,19 +257,57 @@ def gather(self):
         assert result1[0] is not result2[0], key
 
 
-def test_serialization():
-
+def test_torch_serialize():
     module = PyroModule()
     module.x = PyroParam(torch.tensor(1.234), constraints.positive)
+    module.y = nn.Parameter(torch.randn(3))
     assert isinstance(module.x, torch.Tensor)
 
     # Work around https://github.com/pytorch/pytorch/issues/27972
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", category=UserWarning)
         f = io.BytesIO()
         torch.save(module, f)
+        pyro.clear_param_store()
         f.seek(0)
         actual = torch.load(f)
+
+    assert_equal(actual.x, module.x)
+    actual_names = {name for name, _ in actual.named_parameters()}
+    expected_names = {name for name, _ in module.named_parameters()}
+    assert actual_names == expected_names
+
+
+def test_pyro_serialize():
+    class MyModule(PyroModule):
+        def __init__(self):
+            super().__init__()
+            self.x = PyroParam(torch.tensor(1.234), constraints.positive)
+            self.y = nn.Parameter(torch.randn(3))
+
+        def forward(self):
+            return self.x, self.y
+
+    module = MyModule()
+    assert len(pyro.get_param_store()) == 0
+
+    assert isinstance(module.x, torch.Tensor)
+    assert len(pyro.get_param_store()) == 0
+
+    actual = module()  # triggers saving in param store
+    assert_equal(actual[0], module.x)
+    assert_equal(actual[1], module.y)
+    assert len(pyro.get_param_store()) == 2
+
+    # Work around https://github.com/pytorch/pytorch/issues/27972
+    pyro.get_param_store().save("/tmp/pyro_module.pt")
+    pyro.clear_param_store()
+    assert len(pyro.get_param_store()) == 0
+    pyro.get_param_store().load("/tmp/pyro_module.pt")
+    assert len(pyro.get_param_store()) == 2
```

Comment:
Should we also check that the values in the param store before `MyModule` is called matches the module attributes after call? Feel free to disregard this if it is already exercised by some other test.

Reply:
Btw..I'm happy to merge this and resolve these issues in follow-up PRs.

Reply:
done